File: python/baseball_score.py
from bs4 import BeautifulSoup
import requests

def get_baseball():
    URL = "https://www.koreabaseball.com/Schedule/ScoreBoard.aspx"
    page = requests.get(URL)

    soup = BeautifulSoup(page.content, "html.parser")

    r_score = []
    l_score = []
    l_team = []
    r_team = []
    flag = []

    left_team_all = soup.find(id="cphContents_cphContents_cphContents_udpRecord").find_all("p", class_="leftTeam")
    right_team_all = soup.find(id="cphContents_cphContents_cphContents_udpRecord").find_all("p", class_="rightTeam")
    flag_all = soup.find_all("strong",class_="flag")

    for l in left_team_all:
        l_score.append(l.find("em", class_="score").text)
        l_team.append(l.find("strong", class_="teamT").text)
    
    for f in flag_all:
        flag.append(f.text)

    for r in right_team_all:
        r_score.append(r.find("em", class_="score").text)
        r_team.append(r.find("strong", class_="teamT").text)

    l = []
    total = 0
    while total < len(flag):
        l.append(flag[total] + "    " + l_team[total] + " " + str(l_score[total])+" : "+ str(r_score[total])  + " "+ r_team[total] )
        total += 1
    if len(l) == 0:
        msg = "오늘 경기가 없습니다"
    else:
        msg = "\n".join(l)
    return msg

File: go/baseball_score.go
package main

import (
	"fmt"
	"log"
	"net/http"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

type BaseballResult struct {
	Flag   string
	LTeam  string
	LScore string
	RScore string
	RTeam  string
}

func getBaseball() string {
	url := "https://www.koreabaseball.com/Schedule/ScoreBoard.aspx"
	response, err := http.Get(url)
	if err != nil {
		log.Fatal(err)
	}
	defer response.Body.Close()

	doc, err := goquery.NewDocumentFromReader(response.Body)
	if err != nil {
		log.Fatal(err)
	}

	results := make([]BaseballResult, 0)

	doc.Find("#cphContents_cphContents_cphContents_udpRecord .leftTeam").Each(func(i int, s *goquery.Selection) {
		lScore := s.Find("em.score").Text()
		lTeam := s.Find("strong.teamT").Text()

		result := BaseballResult{
			LTeam:  lTeam,
			LScore: lScore,
		}
		results = append(results, result)
	})

	doc.Find("strong.flag").Each(func(i int, s *goquery.Selection) {
		flag := s.Text()
		results[i].Flag = flag
	})

	doc.Find("#cphContents_cphContents_cphContents_udpRecord .rightTeam").Each(func(i int, s *goquery.Selection) {
		rScore := s.Find("em.score").Text()
		rTeam := s.Find("strong.teamT").Text()

		results[i].RScore = rScore
		results[i].RTeam = rTeam
	})

	var msgs []string
	for _, result := range results {
		msg := fmt.Sprintf("%s    %s %s : %s %s", result.Flag, result.LTeam, result.LScore, result.RScore, result.RTeam)
		msgs = append(msgs, msg)
	}
	return strings.Join(msgs, "\n")
}

func main() {
	msg := getBaseball()
	fmt.Println(msg)
}
